torchdrivesim/mesh.py

```diff
@@ -235,28 +236,45 @@ def unpickle(cls, mesh_file_path: str, pickle_module: Any = default_pickle_modul
             return road_mesh
         else:
             raise BadMeshFormat
+    def serialize(self):
+        return {
+            'verts': self.verts.tolist(),
+            'faces': self.faces.tolist()
+        }
 
     def save(self, file_save_path: str):
         """
-        Save the attributes of the object in a non-pickle format.
+        Save the attributes of the mesh object in a json where tensors are converted to lists.
         """
-        if not os.path.exists(os.path.dirname(file_save_path)):
+        directory = os.path.dirname(file_save_path)
+        if directory and not os.path.exists(directory):
             os.makedirs(os.path.dirname(file_save_path))
-        data = {
-            'verts': self.verts.tolist(),
-            'faces': self.faces.tolist()
-        }
-        with open(file_save_path + f'.{self.__class__.__name__}', 'w') as file:
+        data = self.serialize()
+        with open(file_save_path, 'w') as file:
             json.dump(data, file)
 
+    @classmethod
+    def _deserialize_tensors(cls, data: Dict) -> Dict:
+        """
+        Convert list attributes to tensor attributes if there is any tensor attribute
+        """
+        new_data = data.copy()
+        new_data.update(verts=torch.tensor(data['verts']), faces=torch.tensor(data['faces']))
+        return new_data
+
+    @classmethod
+    def deserialize(cls, data: Dict) -> Self:
+        return cls(**cls._deserialize_tensors(data))
+
     @classmethod
     def load(cls, filepath):
         with open(filepath, 'r') as file:
             data = json.load(file)
-        return cls(
-            verts=torch.tensor(data['verts']),
-            faces=torch.tensor(data['faces'])
-        )
+        try:
+            return cls.deserialize(data)
+        except Exception as e:
```

Comment:
This strikes me as overly general, although I'm not sure how to suitably delineate the right exceptions. For sure, however, `json.load` should be inside the `try` block.